src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs

```diff
@@ -68,24 +72,185 @@ internal class ResourceOperation : TypeProvider
+        {
+            var listMethods = GetListMethods(false, true);
+            foreach (var listMethod in listMethods)
+            {
+                var restClientMethod = GetRestClientMethod(listMethod);
+                var pathSegments = restClientMethod?.Request.PathSegments;
 
-        protected virtual IEnumerable<ClientMethod> GetMethodsInScope()
+                // locations scope
+                if (pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/subscriptions/") &&
+                    pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/providers/Microsoft.Compute/locations/") &&
+                    !pathSegments.Any(p => p.Value.IsConstant && (p.Value.Constant.Value?.ToString() == "/resourceGroups/")))
+                {
+                    return listMethod;
+                }
+            }
+
+            return null;
+        }
+
+        private ResourceListMethod? GetResourceGroupsListMethod()
         {
-            return ClientBuilder.BuildMethods(OperationGroup, RestClient, Declaration);
+            var listMethods = GetListMethods(false, true);
+            foreach (var listMethod in listMethods)
+            {
+                var restClientMethod = GetRestClientMethod(listMethod);
+                var pathSegments = restClientMethod?.Request.PathSegments;
+
+                // resourceGroups scope
+                if (pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/resourceGroups/"))
+                {
+                    return listMethod;
+                }
+            }
+
+            return null;
+        }
+
+        private ResourceListMethod? GetTenantListMethod()
+        {
+            var listMethods = GetListMethods(false, true);
+            foreach (var listMethod in listMethods)
+            {
+                var restClientMethod = GetRestClientMethod(listMethod);
+                var pathSegments = restClientMethod?.Request.PathSegments;
+
+                // tenant scope
+                if (!pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/subscriptions/") &&
+                    !pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/resourceGroups/"))
+                {
+                    return listMethod;
+                }
+            }
+
+            return null;
+        }
+
+        public RestClientMethod? GetRestClientMethod(ResourceListMethod? listMethod)
+        {
+            if (listMethod != null)
+            {
+                if (listMethod.PagingMethod != null)
+                {
+                    return listMethod.PagingMethod.Method;
+                }
+                else if (listMethod.ClientMethod != null)
+                {
+                    return listMethod.ClientMethod.RestClientMethod;
+                }
+            }
+
+            return null;
+        }
```

Comment:
This looks more natural if this method belongs to the class `ResourceListMethod` instead of here